rtrlib/rtr/packets.c

```diff
@@ -481,6 +507,27 @@ static bool rtr_pdu_check_size(const struct pdu_header *pdu)
 		if (sizeof(struct pdu_reset_query) == pdu->len)
 			retval = true;
 		break;
+	case ASPA:
+		aspa_pdu = (const struct pdu_aspa *)pdu;
+		min_size = sizeof(struct pdu_aspa);
+		if (aspa_pdu->len < min_size) {
+			RTR_DBG1("PDU is too small to contain ASPA PDU!");
+			break;
+		}
+		
+		// Check if the PDU really contains the ASPA PDU
+		uint16_t asn_count = ntohs(aspa_pdu->provider_count);
+		
+		RTR_DBG("provider_asn_count: %u", asn_count);
+		// ASN is 4 bytes each
+		min_size += asn_count * 4;
```

Comment:
min_size is misleading here and in the following if-statement

Reply:
how'd u call `min_size` instead?

Reply:
i'd suggest something like
```c
int required_size = min_size + asn_count * 4 + ...
````
and maks the debug message something like "has invalid size to contain valid aspa pdu"

Reply:
I think `expected_size` is nicer, because it clearly conveys that this is the size we are expecting from the standard